parquet/src/test/java/org/apache/iceberg/utils/ParameterizedTestExtension.java

```diff
@@ -0,0 +1,255 @@
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+import org.junit.platform.commons.support.AnnotationSupport;
+import org.junit.platform.commons.support.HierarchyTraversalMode;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * This extension is used to implement parameterized tests for Junit 5 to replace Parameterized in
+ * Junit4.
+ *
+ * <p>When use this extension, all tests must be annotated by {@link TestTemplate}.
+ */
+public class ParameterizedTestExtension implements TestTemplateInvocationContextProvider {
+
+    private static final ExtensionContext.Namespace NAMESPACE =
+            ExtensionContext.Namespace.create("parameterized");
+    private static final String PARAMETERS_STORE_KEY = "parameters";
+    private static final String PARAMETER_FIELD_STORE_KEY_PREFIX = "parameterField_";
+    private static final String INDEX_TEMPLATE = "{index}";
+
+    @Override
+    public boolean supportsTestTemplate(ExtensionContext context) {
+        return true;
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(
+            ExtensionContext context) {
+
+        // Search method annotated with @Parameters
+        final List<Method> parameterProviders =
+                AnnotationSupport.findAnnotatedMethods(
+                        context.getRequiredTestClass(),
+                        Parameters.class,
+                        HierarchyTraversalMode.TOP_DOWN);
+        if (parameterProviders.isEmpty()) {
+            throw new IllegalStateException("Cannot find any parameter provider");
+        }
+        if (parameterProviders.size() > 1) {
+            throw new IllegalStateException("Multiple parameter providers are found");
+        }
+
+        Method parameterProvider = parameterProviders.get(0);
+        // Get potential test name
+        String testNameTemplate = parameterProvider.getAnnotation(Parameters.class).name();
+
+        // Get parameter values
+        final Object parameterValues;
+        try {
+            parameterProvider.setAccessible(true);
+            parameterValues = parameterProvider.invoke(null);
+            context.getStore(NAMESPACE).put(PARAMETERS_STORE_KEY, parameterValues);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed to invoke parameter provider", e);
+        }
+        assert parameterValues != null;
```

Comment:
this should be `Preconditions.checkState(..)`

Reply:
Solved in 653fec1f5be055d03c956c4cb274ae015818f2c1